e2e/elasticsearch.go

```diff
@@ -316,6 +316,15 @@ func WaitForNumberOfHits(indexName string, query map[string]interface{}, desired
 	numberOfHits := func() error {
 		hits, err := search(indexName, query)
 		if err != nil {
+			log.WithFields(log.Fields{
+				"desiredHits": desiredHits,
+				"elapsedTime": exp.GetElapsedTime(),
+				"error":       err,
+				"index":       indexName,
+				"retry":       retryCount,
+			}).Warn("There was an error executing the query")
+
+			retryCount++
```

Comment:
Why does the retry counter get incremented here? It doesn't look like it will have an effect, since we just return below it.

Reply:
The `retryCount` variable is defined outside the retry function. This function, which returns an error, will be automatically retried by the backoff library when an error occurs, so that the `retryCount` variable is incremented with each failure.